doc/devs.rst

```diff
@@ -0,0 +1,321 @@
+  public:
+
+    using super = Base;
+
+    using extended_base = my_mixin;
+
+    // ... implementation ...
+
+  private:
+    auto& dref() {
+      return *static_cast<Subtype*>(this);
+    }
+
+    // ... more implementation ...
+  };
+
+By using exactly two template parameters in the order shown above as well as
+defining ``extended_base``, we can use CAF's ``extend`` utility:
+
+.. code-block:: C++
+
+  class my_class : public caf::extend<my_base, my_class>::with<M1, M2, M3> {
+    // ...
+  };
+
+In the example above, ``my_base`` is the base type of our inheritance graph.
+``M1``, ``M2`` and ``M3`` are mixins. The final inheritance graph is
+``my_class`` → ``M1`` → ``M2`` → ``M3`` → ``my_base`` (where A → B means *A
+inherits from B*).
+
+CAF's ``extend`` is of course just syntactic sugar for:
+
+.. code-block:: C++
+
+  class my_class : public M1<M2<M3<my_base, my_class>, my_class>, my_class> {
+    // ...
+  };
+
+Lifting
+~~~~~~~
+
+Broker has a small utility called ``detail::lift`` (implemented in
+``broker/detail/lift.hh``) for lifting member functions into message handlers
+(callbacks). It's sole purpose is to avoid writing repeated lambda expressions
+that only wrap member function calls. Consider this small example:
+
+.. code-block:: C++
+
+  struct calculator {
+    int add(int x, int y) const noexcept { return x + y; }
+
+    int sub(int x, int y) const noexcept { return x - y; }
+
+    // ...
+
+    caf::behavior make_behavior() {
+      using detail::lift;
+      return {
+        lift<atom::add>(*this, calculator::add),
+        lift<atom::sub>(*this, calculator::sub),
+      };
+    }
+  };
+
+By using ``lift``, we avoid repeating the arguments over and over again in
+``make_behavior``. The implementation is equivalent to writing:
+
+.. code-block:: C++
+
+  caf::behavior make_behavior() {
+    return {
+      [this](atom::add, int x, int y) { return add(x, y); },
+      [this](atom::sub, int x, int y) { return sub(x, y); },
+    };
+  }
+
+We can pass any number of template parameters to ``lift`` for prefixing the
+message with atoms or leave the pack empty to dispatch on the member function
+signature only.
+
+Note: lifting overloaded member functions does not work in this concise syntax.
+In order to tell the compiler *which* overload to pick, we need to either store
+the member function pointer in a properly typed variable first or use
+``static_cast``.
+
+The Core Actor
+~~~~~~~~~~~~~~
+
+As the name suggests, this actor embodies the central hub for the
+publish/subscribe communication. Everything flows through this actor: local
+subscriptions, peering requests, local and remote publications, control messages
+for the stores, and so on. However, you might be surprised when looking at
+``core_actor.cc``, as it contains barely any code. Exactly because the core
+actor has so many roles to fill, we have separated it into many functional
+blocks. Most blocks are implemented as mixins_ in order to make each functional
```

Comment:
Per my comment on the PR, a arch diagram would be helpful, in particular how the core + all the mixing ends up looking like in a real (non-unit testing setup)